crates/ruff_python_ast/src/nodes.rs

```diff
@@ -613,6 +623,42 @@ pub enum Expr {
     IpyEscapeCommand(ExprIpyEscapeCommand),
 }
 
+impl Expr {
+    /// Returns `true` if the expression is a literal expression.
+    ///
+    /// A literal expression is either a string literal, bytes literal,
+    /// integer, float, complex number, boolean, `None`, or ellipsis (`...`).
+    pub fn is_literal_expr(&self) -> bool {
+        matches!(
+            self,
+            Expr::StringLiteral(_)
+                | Expr::BytesLiteral(_)
+                | Expr::NumberLiteral(_)
+                | Expr::BooleanLiteral(_)
+                | Expr::NoneLiteral(_)
+                | Expr::EllipsisLiteral(_)
+        )
+    }
+
+    pub fn is_implicit_concatenated_string(&self) -> bool {
```

Comment:
This is temporary and will be removed with the addition of the new node for implicit string concatenation.